ALGORITHMS FOR DNA SEQUENCING

This file contains codes created for completing the various tasks required to complete the course

WEEK 1
1. Basics of working with String & manipulating DNA strands as string such as finding the reverse complement of the string
2. Algorithms for parsing a genome from a 'fasta' file and sequencing reads from a 'fastq' file
3. Creating charts to observe the number of time a quality score was found in a sequencing read & observing the GC content per base. Furthermore, algorithms for conversion of quality score to ASCII & vice-versa was written.
4. Matching artificial reads to a genome
5. Matching real reads to a genome. Warning signs such as change in reads due to sequencing errors, double-stranded nature of DNA was also taken into considerations when matching the reads.

WEEK 2
1. Naive Algorithm of finding a pattern 'p' in text 't'
2. Boyer-Moore Algorithm, a more complex algorithm for finding a pattern 'p' in text 't'. The algorithm is able to skip multiple iterations unlike naive algorithm
3. Preprocessing the text 't' to aid in finding a pattern 'p' in 't'. Main method done this week was by using a ordered indexing algorithm.
4. Changing the Boyer-Moore algorithm & Indexing algorithm to account for 'd' no. of mismatches using the pidgeom hole principle.